auraed/src/cells/cell_service/cells/cell.rs

```diff
@@ -229,6 +233,28 @@ impl CellsCache for Cell {
 
         children.broadcast_kill()
     }
+
+    fn cell_graph(&self, node: &GraphNode) -> GraphNode {
+        let cgroup = self.spec().cgroup_spec.clone();
+        let iso_ctl = self.spec().iso_ctl.clone();
+
+        let cell_info = CellInfo {
+            cell_name: self.cell_name.clone(),
+            cpu: cgroup.cpu,
+            cpuset: cgroup.cpuset,
+            isolate_network: iso_ctl.isolate_network,
+            isolate_process: iso_ctl.isolate_process,
+        };
+
+        let CellState::Allocated { children, .. } = &self.state else {
+            return GraphNode {
```

Comment:
Instead of returning a node without children, this should result in `Err(CellsError::CellNotAllocated { ... })`. The only way to hit this code is when the cache has gotten out of sync with the system. A call to `Cell` that results in `CellNotAllocated` can be used by `Cells` as a signal to remove the `Cell` from the cache and then handle the situation appropriately. For `cell_graph` I think the appropriate action is to continue to the next cell after removing the current one from the cache.

Reply:
"out of sync" as in the `cgroup` doesn't exist anymore?

Reply:
Or are there other failure modes? Kind of want to write a test for this to ensure invalid cells get properly removed in this scenario. Do we have any examples of such tests?

Reply:
It more likely means the cgroup never existed. If you look at `allocate` on `Cells`, you can see that we insert a `Cell` in to the cache and then call `cell.allocate()?`, which could fail. If it fails, then you have a cell in the cache that is not in the `Allocated` state.

Reply:
Also in `Cells`, you can see how `get` and `get_mut` react when encoutering `CellNotAllocated` (they remove the cell from the cache).

Reply:
Thinking about this a little more. If `cells.allocate()?` fails, wouldn't it make sense to remove that cell from the cache immediately after that call in `Cells::allocate`?

Reply:
If I remember correctly, I fought the borrow checker and lost trying to do that and then didn't go back to try again :).

Reply:
Ha ok. I left a comment for now to remind us in the future. I spent some time trying to write a test that would cause `allocate()?` to fail, but I was not able to do so. 😒